server_test.go

```diff
@@ -3350,6 +3350,133 @@ func TestMaxBodySizePerRequest(t *testing.T) {
 	}
 }
 
+func TestStreamRequestBody(t *testing.T) {
+	t.Parallel()
+
+	part1 := strings.Repeat("1", 1<<10)
+	part2 := strings.Repeat("2", 1<<20-1<<10)
+	contentLength := len(part1) + len(part2)
+	next := make(chan struct{})
+	done := make(chan struct{})
+
+	s := &Server{
+		Handler: func(ctx *RequestCtx) {
+			checkReader(t, ctx.RequestBodyStream(), part1)
+			close(next)
+			<-done
```

Comment:
Instead of adding the `done` channel I think it's probably a better idea to add a `sync.Mutex` to `readWriter` to make its operations safe just like reading and writing on a normal socket would be.

Reply:
I think we can't use `sync.Mutex`. If handler get lock first, then second part can't be sent, and the test case will timeout.
So we should use a more real `Conn`. Let me try it.

Reply:
Use `fasthttputil.NewPipeConns()` now.